securecookie.go

```diff
@@ -266,9 +278,9 @@ func verifyMac(h hash.Hash, value []byte, mac []byte) error {
 // encrypt encrypts a value using the given block in counter mode.
 //
 // A random initialization vector (http://goo.gl/zF67k) with the length of the
-// block size is prepended to the resulting ciphertext.
-func encrypt(block cipher.Block, value []byte) ([]byte, error) {
-	iv := GenerateRandomKey(block.BlockSize())
+// block size is read from the given reader and prepended to the resulting ciphertext.
+func encrypt(block cipher.Block, value []byte, randomReader io.Reader) ([]byte, error) {
+	iv := GenerateRandomKeyReader(block.BlockSize(), randomReader)
```

Comment:
If you make `GenerateRandomKey` use `s.randomReader`, you can avoid passing `randomReader` to `encrypt` and just use it.